### PR TITLE
[5.3] [Draft] Introduce Illuminate\Notifications\Message (with breaking changes)

### PR DESCRIPTION
**Pull request by crynobone**
This would allow user to customize message for each notifiable.

The main idea with this is to allow user to customise message send to each users (`notifiables`). Without this we can only send generic message for every users. This also clean-up some of the old unusable code from the last notifications refactors.

Signed-off-by: crynobone crynobone@gmail.com

### CODE REVIEW

File: src/Illuminate/Auth/Notifications/ResetPassword.php [hunk 1]
```diff
@@ -38,15 +39,16 @@ public function via($notifiable)
...
-    public function message()
+    public function message(Message $message)
     {
-        return $this->line([
+        return $message->line([
                 'You are receiving this email because we received a password reset request for your account.',
                 'Click the button below to reset your password:',
             ])
-            ->action('Reset Password', url('password/reset', $this->token))
+            ->action('Reset Password', url('password/reset', $this->token).'?email='.urlencode($message->notifiable->email))
```
Comment (crynobone) [inline src/Illuminate/Auth/Notifications/ResetPassword.php]: We can either send `(Message $message)` or `(Message $message, $notifiable)`

File: src/Illuminate/Notifications/Message.php [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+     *
+     * @return array
+     */
+    public function toArray()
+    {
+        return [
+            'notifiable' => $this->notifiable,
+            'application' => $this->notification->application,
+            'logoUrl' => $this->notification->logoUrl,
+            'level' => $this->notification->level,
```
Comment (crynobone) [inline src/Illuminate/Notifications/Message.php]: `application`, `logoUrl` and `level` most likely going to be shared for a single Notification. so might as well do this.

**Comment by crynobone** [comment]
I'm marking this as a draft so @taylorotwell (and everyone) else could look at and suggest feedback first.

**Comment by crynobone** [comment]
Just rebased to latest code, need to fix the failing test.

**Comment by taylorotwell** [comment]
I was able to introduce this same concept in a slightly different way. Check out my latest commits and let me know what you think.

**Comment by crynobone** [comment]
That looks awesome ❤️

**Pull request closed without merging**